tests/xharness/TestReporter.cs

```diff
@@ -0,0 +1,490 @@
+						runner.Logs,
+						"tcp-connection",
+						runner.AppInformation.AppName,
+						runner.AppInformation.Variation,
+						$"TcpConnection on {deviceName}",
+						$"Device {deviceName} could not reach the host over tcp.",
+						runner.MainLog.FullPath,
+						runner.XmlJargon);
+				}
+			} else if (timedout) {
+				ResultParser.GenerateFailure (
+					runner.Logs,
+					"timeout",
+					runner.AppInformation.AppName,
+					runner.AppInformation.Variation,
+					$"App Timeout {runner.AppInformation.AppName} {runner.AppInformation.Variation} on bot {deviceName}",
+					$"{runner.AppInformation.AppName} {runner.AppInformation.Variation} Test run timed out after {Timeout.TotalMinutes} minute(s) on bot {deviceName}.",
+					runner.MainLog.FullPath,
+					runner.XmlJargon);
+			}
+		}
+
+		public async Task<(TestExecutingResult ExecutingResult, string FailureMessage)> ParseResult ()
+		{
+			var result = (ExecutingResult: TestExecutingResult.Finished, FailureMessage: (string) null);
+			var crashed = false;
+			if (File.Exists (listener.TestLog.FullPath)) {
+				WrenchLog.WriteLine ("AddFile: {0}", listener.TestLog.FullPath);
+				(Success, crashed) = await TestsSucceeded (runner.AppInformation, listener.TestLog.FullPath, timedout);
+			} else if (timedout) {
+				WrenchLog.WriteLine ("AddSummary: <b><i>{0} never launched</i></b><br/>", runner.RunMode);
+				runner.MainLog.WriteLine ("Test run never launched");
+				Success = false;
+			} else if (launchFailure) {
+ 				WrenchLog.WriteLine ("AddSummary: <b><i>{0} failed to launch</i></b><br/>", runner.RunMode);
+ 				runner.MainLog.WriteLine ("Test run failed to launch");
+ 				Success = false;
+			} else {
+				WrenchLog.WriteLine ("AddSummary: <b><i>{0} crashed at startup (no log)</i></b><br/>", runner.RunMode);
+				runner.MainLog.WriteLine ("Test run crashed before it started (no log file produced)");
+				crashed = true;
+				Success = false;
+			}
+				
+			if (!Success.HasValue)
+				Success = false;
+
+			var crashLogWaitTime = 0;
+			if (!Success.Value)
+				crashLogWaitTime = 5;
+			if (crashed)
+				crashLogWaitTime = 30;
+
+			await crashReporter.EndCaptureAsync (TimeSpan.FromSeconds (crashLogWaitTime));
+
+			if (timedout) {
+				result.ExecutingResult = TestExecutingResult.TimedOut;
+			} else if (crashed) {
+				result.ExecutingResult = TestExecutingResult.Crashed;
+			} else if (Success.Value) {
+				result.ExecutingResult = TestExecutingResult.Succeeded;
+			} else {
+				result.ExecutingResult = TestExecutingResult.Failed;
+			}
+
+			// Check crash reports to see if any of them explains why the test run crashed.
+			if (!Success.Value) {
+				int pid = -1;
+				string crashReason = null;
+				foreach (var crashLog in crashLogs) {
+					try {
+						runner.Logs.Add (crashLog);
+
+						if (pid == -1) {
+							// Find the pid
+							pid = await GetPidFromMainLog ();
+						}
+
+						GetCrashReason (pid, crashLog, out crashReason);
+						if (crashReason != null) 
```

Comment:
@spouliot in this line is where we have the fix for the false positives. We used to create a crash report, but we are going through the reports even when we did not crash (adding them is not bad practice). But doing the crash xml failure meant a diff failure count. Everything got moved to GenerateXmlFailures which takes a param stating if there was a crash or not.